### ISSUE TITLE
wrong line reported for no-extra-parens

### ISSUE THREAD

**Issue by remitbri** [eslint/eslint#1814]
What about allowing parentheses after a return in the render method of a component, like in
```
render() {
    return (
        <div>
            <h1>{this.props.title}</h1>
            <form>
            // and so on
            </form>
        </div>
    )
}
```
even with no-extra-parens switched on? or as an option?
Granted, it's for better reading and code style perhaps, but on the other hand, it's more useful than in the case of `a = (b * c)`, `(a * b) + c` or `typeof (a)`…

As opposed to having to write
```
render() {
    return <div>
        <h1>{this.props.title}</h1>
        <form>
        // and so on
        </form>
    </div>
}
```

**Comment by nzakas** [comment]
Sorry, I'm not following. Are you reporting a bug? Or an issue with a rule?

**Comment by remitbri** [comment]
Hmm, I think it's a bug. 
But I'm not 100% affirmative. I could see why my example could still be regarded as "unnecessary use of parentheses"…

**Comment by nzakas** [comment]
It's really helpful if you can phrase your description as:

1. What you did
2. What you expected to happen
3. What actually happened (including console output)

I'm having a hard time understanding what you're asking for.

**Comment by remitbri** [comment]
Sorry. So:
.eslintrc
```
{
  "env" : {
    "node": true,
  },
  "ecmaFeatures" : {
    "jsx": true
  },
  "rules" : {
    "no-extra-parens": 2
  }
}
```

my jsx file
```jsx
"use strict";
var React = require("react");
var MyApp = React.createClass({
  render: function() {
      return (
          <div>
              <h1>{this.props.title}</h1>
              <form>
              // and so on
              </form>
          </div>
      );
  }
});
module.exports = MyApp;
```
The error :
```
  6:10  error  Gratuitous parentheses around expression  no-extra-parens
```

I think this is a bug: 
1) parentheses are necessary for this coding style, and this is not a matter of cosmetics like in `(a * b) + c` or `typeof (a)`
2) the error mentions 6:10, which is where the `<div>` starts, not where the opening parenthesis is (a line before)

It could be not a bug if it's decided that no, either I write ```return <div>``` or I switch the no-extra-parens rule off.

**Comment by nzakas** [comment]
Thanks. The only bug I see here is the line number that was reported. The rule is designed to warn when parentheses are used but not required, so it is behaving correctly in this situation. We can fix the line number, but if you want it to not warn at all, you'll need to disable the rule.

**Comment by remitbri** [comment]
The not-a-bug one: thanks, that's what I suspected —and why I asked.
The indeed-a-bug one: yeah, another one detected!

**Comment by nzakas** [comment]
Working on this.
